bundles/org.openhab.binding.ferroamp/src/main/java/org/openhab/binding/ferroamp/internal/FerroampHandler.java

```diff
@@ -0,0 +1,255 @@
+/**
+ * Copyright (c) 2010-2025 Contributors to the openHAB project
+ *
+ * See the NOTICE file(s) distributed with this work for additional
+ * information.
+ *
+ * This program and the accompanying materials are made available under the
+ * terms of the Eclipse Public License 2.0 which is available at
+ * http://www.eclipse.org/legal/epl-2.0
+ *
+ * SPDX-License-Identifier: EPL-2.0
+ */
+package org.openhab.binding.ferroamp.internal;
+
+import java.util.ArrayList;
+import java.util.List;
+import java.util.Objects;
+import java.util.UUID;
+import java.util.concurrent.TimeUnit;
+
+import org.eclipse.jdt.annotation.NonNullByDefault;
+import org.eclipse.jdt.annotation.Nullable;
+import org.openhab.core.io.transport.mqtt.MqttBrokerConnection;
+import org.openhab.core.io.transport.mqtt.MqttMessageSubscriber;
+import org.openhab.core.library.types.StringType;
+import org.openhab.core.thing.ChannelUID;
+import org.openhab.core.thing.Thing;
+import org.openhab.core.thing.ThingStatus;
+import org.openhab.core.thing.ThingStatusDetail;
+import org.openhab.core.thing.binding.BaseThingHandler;
+import org.openhab.core.types.Command;
+import org.openhab.core.types.State;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+/**
+ * The {@link FerroampHandler} is responsible for handling of the values sent to and from the binding.
+ *
+ * @author Örjan Backsell - Initial contribution
+ *
+ */
+
+@NonNullByDefault
+public class FerroampHandler extends BaseThingHandler implements MqttMessageSubscriber {
+    private final static Logger logger = LoggerFactory.getLogger(FerroampHandler.class);
+    private @Nullable static MqttBrokerConnection ferroampConnection;
+    FerroampMqttCommunication ferroampMqttCommunication = new FerroampMqttCommunication(thing);
+    FerroampConfiguration ferroampConfig = new FerroampConfiguration();
+
+    private static List<FerroampChannelConfiguration> channelConfigEhub = new ArrayList<>();
+    private static List<FerroampChannelConfiguration> channelConfigSsoS1 = new ArrayList<>();
+    private static List<FerroampChannelConfiguration> channelConfigSsoS2 = new ArrayList<>();
+    private static List<FerroampChannelConfiguration> channelConfigSsoS3 = new ArrayList<>();
+    private static List<FerroampChannelConfiguration> channelConfigSsoS4 = new ArrayList<>();
+    private static List<FerroampChannelConfiguration> channelConfigEso = new ArrayList<>();
+    private static List<FerroampChannelConfiguration> channelConfigEsm = new ArrayList<>();
+
+    long refreshInterval = 30;
```

Comment:
Moving the `refreshInterval` to the FerrampConfiguration (and adding it to the thing-type.xml (mark it as advanced) and readme.md) makes this configurable for users.

Reply:
20250319
Added refreshInterval to FerrampConfiguration, thing-type  and README.md